Add annotated extension and service ontologies

birkland opened the pull request:
There are a few suggested changes and open questions related to the ontologies in the [draft google doc](https://docs.google.com/document/d/1p1pMqVt8XjK7CJ7Ep_SXLV_AitgI9dkUTU9_nG1Lpy4/edit).  We can create new issues for anything that can't or shouldn't be immediately addressed in this PR.

acoburn commented on fcrepo-api-x-model/src/main/resources/ontologies/apix-extension.ttl:
```diff
@@ -0,0 +1,65 @@
+apix:Extension a owl:Class;
+    rdfs:label "API-X extension"@en;
+    rdfs:comment
+    "Relates Services to Repository objects. May expose a service on repository resources (‘exposing’ modality), or modify request or responses to repository objects (‘intercepting’ modality)."@en .
```
"smart quotes"? I don't have a strong opinion, but I would have expected just a regular old single quote i.e.: '

birkland replied:
Ha, That was a cut 'n paste from the rendered markdown - I didn't realize it did that.

ajs6f replied:
Pretty sure you set Checkstyles to catch that kind of thing.

birkland replied:
Hm, the following didn't work:
Add to fcrepo-build-tools/src/main/resources/fcrepo-checkstyle/checkstyle.xml

<pre>
  &lt;module name="RegexpSingleline"&gt;
    &lt;property name="fileExtensions" value="java,css,js,xml,ttl"/&gt;
    &lt;property name="format" value=".*[“‘’”]+.*"/&gt;
    &lt;property name="message" value="Found smart quotes"/&gt;
  &lt;/module&gt;
</pre>

ajs6f replied:
Is the regexp too greedy? Maybe `.*?`?

birkland replied:
No, `.*?` didn't work.  The regex works fine manually.  Enabling debug logging doesn't reveal useful information.  I've confirmed that checkstyle truly is using the modified file, but have no way of confirming that it is actually executing the `RexepSingleline` in the first place, nor a way of confirming if checkstyle is looking at the `.ttl` file in the first place outside of attaching a debugger to see what's really going on.

So I'm just going to fix the smart quotes with a commit, then maybe add a jira issue to add smart quote checking to fcrepo-build-tools

emetsger commented on fcrepo-api-x-model/src/main/resources/ontologies/apix-extension.ttl:
```diff
@@ -0,0 +1,65 @@
+apix:Extension a owl:Class;
+    rdfs:label "API-X extension"@en;
+    rdfs:comment
+    "Relates Services to Repository objects. May expose a service on repository resources (‘exposing’ modality), or modify request or responses to repository objects (‘intercepting’ modality)."@en .
```
Yeah google does it too.  It's bitten me multiple times!

On Tue, Aug 30, 2016 at 3:33 PM, birkland notifications@github.com wrote:

> In fcrepo-api-x-model/src/main/resources/ontologies/apix-extension.ttl
> https://github.com/fcrepo4-labs/fcrepo-api-x/pull/38#discussion_r76863849
> :
> 
> > @@ -0,0 +1,65 @@
> > +@prefix apix:http://fedora.info/definitions/v4/api-extension# .
> > +@prefix fcr:http://fedora.info/definitions/v4/repository# .
> > +@prefix owl:http://www.w3.org/2002/07/owl# .
> > +@prefix rdfs:http://www.w3.org/2000/01/rdf-schema# .
> > +@prefix svc:http://fedora.info/definitions/v4/service# .
> > +@prefix xsd:http://www.w3.org/2001/XMLSchema# .
> > +
> > +apix: a owl:Ontology;
> > -    owl:imports svc:;
> > -    rdfs:label "API Extension ontology" .
> >   +
> >   +apix:Extension a owl:Class;
> > -    rdfs:label "API-X extension"@en;
> > -    rdfs:comment
> > -    "Relates Services to Repository objects. May expose a service on repository resources (‘exposing’ modality), or modify request or responses to repository objects (‘intercepting’ modality)."@en .
> 
> Ha, That was a cut 'n paste from the rendered markdown - I didn't realize
> it did that.
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> https://github.com/fcrepo4-labs/fcrepo-api-x/pull/38/files/29c401b3651622a342cdfd6ca7183566ce11d612#r76863849,
> or mute the thread
> https://github.com/notifications/unsubscribe-auth/AAI-Gg1vw3H-BbISUBwRa35EU0cy3XKQks5qlIWNgaJpZM4Jw3ji
> .

birkland commented:
Rebased on top of latest master, and moved ontology .ttl files into their own module.  Shall we merge this, then create another issue for producing rdf/xml derivatives as an artifact?
